### PR TITLE
fixes #69

### LINKED ISSUE #69: add footer
**Describe the solution you'd like**
A small footer with a nav to legal pages(currently imprint) and a copyright notice(Björn Büttner and contributors)

### PR DESCRIPTION
**Pull request by MohamedSy1**

### CODE REVIEW

File: website/src/index.css [hunk 1]
```diff
@@ -150,6 +159,23 @@ a[target="_blank"] {
 }
 
 @media only screen and (width >= 768px) {
+  footer {
+    display: flex;
+    padding: 1rem 0;
+    margin-top: auto;
+    background-color: var(--dark-gray);
```
Comment (Idrinth) [inline website/src/index.css:166]: please make it light gray as the header

File: website/src/index.css [hunk 2]
```diff
@@ -7,9 +8,17 @@
...
+html {
+  min-height: 100%;
+}
+
 body {
   font-family: Roboto, sans-serif;
   font-size: 16px;
+  height: 100%;
+  padding-bottom: 3em;
+  display: flex;
```
Comment (Idrinth) [inline website/src/index.css:20]: why is flex needed here?

File: website/src/components/footer.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      <a href="#" >About</a>
+    </li>
+    <li>
+      <a href="#" >Privacy Policy</a>
+    </li>
+    <li>
+      <a href="#" >Licensing</a>
+    </li>
+    <li>
+      <a href="/imprint" >Imprint</a>
```
Comment (Idrinth) [inline website/src/components/footer.tsx:16]: please remove the unneeded space in the a tag

File: website/src/components/footer.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+const Footer = () => <footer>
+  <span className='footer-text'>© 2020-2024 Björn Büttner and contributors.</span>
+  <ul id="policy-list">
```
Comment (Idrinth) [inline website/src/components/footer.tsx:5]: no need for the id, see below

File: website/src/components/footer.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+const Footer = () => <footer>
+  <span className='footer-text'>© 2020-2024 Björn Büttner and contributors.</span>
```
Comment (Idrinth) [inline website/src/components/footer.tsx:4]: no need for the class, see below

File: website/src/main.tsx [hunk 6]
```diff
@@ -33,6 +34,7 @@ ReactDOM.createRoot(document.getElementById('root',)!,).render(
             <>
               <Navbar />
               <Home />
+              <Footer />
```
Comment (Idrinth) [inline website/src/main.tsx:37]: the footer is required on all pages, not just home

File: website/src/pages/home/index.tsx [hunk 7]
```diff
@@ -51,5 +51,6 @@ const Index = () => <section className="home">
         complicated validations or huge response bodies on the data gathering.
     </p>
   </div>
+
```
Comment (Idrinth) [inline website/src/pages/home/index.tsx:54]: this empty line doesn't look intentional.

**Review by Idrinth — CHANGES_REQUESTED** [review]
Thanks for helping out :) I have a few questions and requests, that I would ask you to answer or fix

File: website/src/components/footer.tsx (hunk 3 above)
Comment (MohamedSy1) [reply website/src/components/footer.tsx:16]: Done

File: website/src/index.css (hunk 2 above)
Comment (MohamedSy1) [reply website/src/index.css:20]: I was trying something out and forgot to delete but it does nothing. I removed it.

File: website/src/index.css [hunk 8]
```diff
@@ -150,6 +159,23 @@ a[target="_blank"] {
 }
 
 @media only screen and (width >= 768px) {
+  footer {
+    display: flex;
+    padding: 1rem 0;
+    margin-top: auto;
```
[the remark this answers is not in the document]
Comment (MohamedSy1) [reply website/src/index.css:165]: Removed margin auto

File: website/src/index.css [hunk 9]
```diff
@@ -150,6 +159,23 @@ a[target="_blank"] {
...
+    background-color: var(--dark-gray);
+    position: absolute;
+    bottom: 0;
+    width: 100%;
+  }
+
+  #policy-list {
+    display: flex;
+    justify-content: flex-end;
+    position: fixed;
```
[the remark this answers is not in the document]
Comment (MohamedSy1) [reply website/src/index.css:175]: it is needed if i delete it the ul only show half the page

File: website/src/index.css [hunk 10]
```diff
@@ -150,6 +159,23 @@ a[target="_blank"] {
...
+    min-width: 95%;
```
[the remark this answers is not in the document]
Comment (MohamedSy1) [reply website/src/index.css:176]: I did 95% so there could little spacing to the right

File: website/src/index.css [hunk 11]
```diff
@@ -183,10 +209,37 @@ a[target="_blank"] {
...
   }
 
-  nav ul ul {
+  footer > ul > li > a {
+    text-decoration: none;
+    color: white;
+  }
+
+  footer > ul > li {
+    display: flex;
```
[the remark this answers is not in the document]
Comment (MohamedSy1) [reply website/src/index.css:218]: for some reason, the unordered list dots show up if I remove the display flex so I leave it there.

**Review by MohamedSy1 — COMMENTED** [review]
Completed and implemented feedback. Great feedback.

File: website/src/index.css (hunk 11 above)
[the remark this answers is not in the document]
Comment (Idrinth) [reply website/src/index.css:218]: that is handled with `list-style: none` usually :)

File: website/src/index.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (Idrinth) [reply website/src/index.css:176]: width 100% and padding-right 5% would be the usual solution

**Comment by Idrinth** [comment]
please don't forget to add a simple test for the component, otherwise coveralls will fail :) have a look at the navbar test, it doesn't have to be more complicated

**Comment by Idrinth** [comment]
@MohamedSy1 please solve the merge conflicts :)

File: website/package.json [hunk 12]
```diff
@@ -25,7 +25,8 @@
     "@babel/plugin-syntax-import-attributes": "^7.23.3",
     "@testing-library/react": "^14.1.2",
     "@types/chai": "^4.3.11",
-    "@types/mocha": "^10.0.5",
+    "@types/jest": "^29.5.11",
```
Comment (Idrinth) [inline website/package.json:28]: why is jest here? We are running tests with mocha and chai

File: website/src/index.css [hunk 13]
```diff
@@ -7,9 +7,16 @@
...
+
+html {
+  height: 100vh;
+}
+
 body {
   font-family: Roboto, sans-serif;
   font-size: 16px;
+  min-height: 100vh;
+  /* padding-bottom: 3em; */
```
Comment (Idrinth) [inline website/src/index.css:19]: remove comment

File: website/src/index.css [hunk 14]
```diff
@@ -149,7 +156,33 @@ a[target="_blank"] {
...
+
+section {
+  min-height: calc(100vh - 6em - 2rem - 3px);
+}
+
 @media only screen and (width >= 768px) {
+  footer {
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    /* position: fixed; */
```
Comment (Idrinth) [inline website/src/index.css:171]: remove comment

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: website/package.json [hunk 15]
```diff
@@ -25,7 +25,8 @@
     "@babel/plugin-syntax-import-attributes": "^7.23.3",
     "@testing-library/react": "^14.1.2",
```
Comment (Idrinth) [inline website/package.json:26]: this is not used to my knowledge

File: website/src/index.css [hunk 16]
```diff
@@ -149,7 +156,33 @@ a[target="_blank"] {
...
+#root {
+  min-height: 100vh;
+}
+
+section {
+  min-height: calc(100vh - 6em - 2rem - 3px);
+}
+
 @media only screen and (width >= 768px) {
+  footer {
```
Comment (Idrinth) [inline website/src/index.css:168]: the footer is unstyled for mobile, please move some of the styling there.

File: website/src/components/footer.tsx [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+  <span >© 2020-2024 Björn Büttner and contributors.</span>
+  <ul>
+    <li>
+      <a href="#">About</a>
+    </li>
+    <li>
+      <a href="#">Privacy Policy</a>
+    </li>
+    <li>
+      <a href="#">Licensing</a>
```
Comment (Idrinth) [inline website/src/components/footer.tsx:13]: License goes to /license and uses the content of LICENSE in the root of the repository

File: website/src/components/footer.tsx [hunk 18]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+const Footer = () => <footer>
+  <span >© 2020-2024 Björn Büttner and contributors.</span>
+  <ul>
+    <li>
+      <a href="#">About</a>
+    </li>
+    <li>
+      <a href="#">Privacy Policy</a>
```
Comment (Idrinth) [inline website/src/components/footer.tsx:10]: doesn't exist, it's part of the imprint

File: website/src/components/footer.tsx [hunk 19]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+const Footer = () => <footer>
+  <span >© 2020-2024 Björn Büttner and contributors.</span>
+  <ul>
+    <li>
+      <a href="#">About</a>
```
Comment (Idrinth) [inline website/src/components/footer.tsx:7]: doesn't exist

File: website/src/components/footer.tsx [hunk 20]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+const Footer = () => <footer>
+  <span >© 2020-2024 Björn Büttner and contributors.</span>
```
Comment (Idrinth) [inline website/src/components/footer.tsx:4]: useless space in span

**Review by Idrinth — CHANGES_REQUESTED** [review]

File: website/src/index.css (hunk 16 above)
Comment (Idrinth) [reply website/src/index.css:168]: grafik

**Comment by Idrinth** [comment]
/usage/logging has the footer missplaced
grafik

File: website/package.json (hunk 12 above)
Comment (MohamedSy1) [reply website/package.json:28]: I am not sure how it got there and don't know what it does.

File: website/package.json (hunk 15 above)
Comment (MohamedSy1) [reply website/package.json:26]: I have never worked with a package.json. I would like for you to help me understand.

File: website/src/index.css (hunk 16 above)
Comment (MohamedSy1) [reply website/src/index.css:168]: Not sure what you mean do I have to create a media query for mobile version.

Comment (Idrinth) [reply website/src/index.css:166]: no, mobile is the media-query-less default

**Review by Idrinth — APPROVED** [review]

**Pull request merged**